lnwallet/channel.go

```diff
@@ -1180,13 +1203,8 @@ func newBreachRetribution(chanState *channeldb.OpenChannel, stateNum uint64,
 	// If the local balance exceeds the remote party's dust limit,
 	// instantiate the local sign descriptor.
 	if localAmt >= chanState.RemoteChanCfg.DustLimit {
-		// We'll need to reconstruct the single tweak so we can sweep
-		// our non-delayed pay-to-self output self.
-		singleTweak := SingleTweakBytes(commitmentPoint,
-			chanState.LocalChanCfg.PaymentBasePoint)
-
 		localSignDesc = &SignDescriptor{
-			SingleTweak:   singleTweak,
+			SingleTweak:   keys.localKeyTweak,
```

Comment:
Note that the tweak used is different depending on exactly which key is being used to sign. We recently fixed a bug which was caused by assuming the tweak is the same for each key within a commitment transaction: 30d5b9062617d003b1f6c83f3c576a8efc57b0c1

The `commitKeyRing` struct should be expanded to define tweaks for both the `delayKey` and the `localKey`.

Reply:
Yes both need to be added as when sweeping outgoing HTLC's, we need _both_ tweaks: local to generate the signature for the multi-sig script and delay when we're sweeping the HTLC output itself.